File: model/src/main/kotlin/net/dungeonhub/model/cnt_request/CntRequestUpdateModel.java
package net.dungeonhub.model.cnt_request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;
import me.taubsie.dungeonhub.common.DungeonHubService;
import me.taubsie.dungeonhub.common.entity.model.UpdateModel;
import me.taubsie.dungeonhub.common.model.discord_user.DiscordUserModel;

import javax.annotation.Nullable;

@Getter
@Setter
@AllArgsConstructor
@Builder
public class CntRequestUpdateModel implements UpdateModel<CntRequestModel> {
    //TODO discuss -> needs to be changable?
    //private long messageId;
    @Nullable
    private DiscordUserModel claimer;
    private boolean removeClaimer;
    @Nullable
    private String coinValue;
    @Nullable
    private String description;
    @Nullable
    private String requirement;

    @Override
    public CntRequestModel apply(CntRequestModel model) {
        if(removeClaimer) {
            model.setClaimer(null);
        }

        if(claimer != null) {
            model.setClaimer(claimer);
        }

        if(coinValue != null) {
            model.setCoinValue(coinValue);
        }

        if(description != null) {
            model.setDescription(description);
        }

        if(requirement != null) {
            model.setRequirement(requirement);
        }

        return model;
    }

    public String toJson() {
        return DungeonHubService.getInstance().getGson().toJson(this);
    }
}

File: model/src/main/kotlin/net/dungeonhub/model/cnt_request/CntRequestUpdateModel.kt
package net.dungeonhub.model.cnt_request

import net.dungeonhub.model.discord_user.DiscordUserModel
import net.dungeonhub.service.MoshiService
import net.dungeonhub.structure.model.UpdateModel

class CntRequestUpdateModel(
    claimer: DiscordUserModel?,
    var coinValue: String?,
    var description: String?,
    var requirement: String?,
    var completed: Boolean?
) : UpdateModel<CntRequestModel> {
    var claimer = claimer
        set(value) {
            resetClaimer = value == null
            field = value
        }

    var resetClaimer = false
        private set

    fun toJson(): String {
        return MoshiService.moshi.adapter(CntRequestUpdateModel::class.java).toJson(this)
    }
}